### PR TITLE
Addressed design notes and added x and y axis formatter functions

### PR DESCRIPTION
**Pull request by dalerasrorov-eb**
More improvements for Scatter Plot

## Description
* Added 0.24 opacity, noted by Sun that it's used in Stacked Area and other charts.
* `circleOpacity` sets `fill-opacity` rather than `opacity`. The outline color should be solid -- this follows Sun's notes as well
* Added tests for data points
* added `xAxisFormat` and `yAxisFormat` chart API methods for formatting x and y axis values
* Used correct class for both axis labels
* Added DOM tests for the new API functions

## Motivation and Context
Improvements and essential API methods

## How Has This Been Tested?
Added multiple tests to `scatter-plot.spec.js`
* Added tests for data points
* Added DOM tests for the new API functions

## Screenshots (if appropriate):
* Circle opacity default is 0.24 and stroke (outline) color is solid:

screen shot 2018-03-30 at 12 47 00 am

screen shot 2018-03-30 at 12 47 04 am

## Types of changes

- [ ] Refactor (changes the way we code something without changing its functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the [code style guide](https://github.com/eventbrite/britecharts/blob/master/CODESTYLEGUIDE.md) of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/charts/scatter-plot.js [hunk 1]
```diff
@@ -713,6 +717,21 @@ define(function(require) {
             return this;
         };
 
+        /**
+         * Exposes ability to set the format of x-axis values
+         * @param  {String} _x               Desired height for the chart
```
Comment (Golodhros) [inline src/charts/scatter-plot.js]: Is this working fine with a format of ' ' as default?

Comment (DalerAsrorov) [reply src/charts/scatter-plot.js]: Yes! I believe ‘’ let’s it figure out the format on its own. But I tried null or undefined and it was throwing an error. If there is some other better ways you know, we can do that instead!

Comment (Golodhros) [reply src/charts/scatter-plot.js]: Awesome, sounds good!

**Comment by miglesiasEB** [comment]
LGTM

**Pull request merged**